test/definitions/native.jl

```diff
@@ -250,7 +250,7 @@ module LazyCodegen
 
     import GPUCompiler: deferred_codegen_jobs
     @generated function deferred_codegen(::Val{f}, ::Val{tt}) where {f,tt}
-        job, _ = native_job(f, tt)
+        job, _ = native_job(typeof(f), tt)
```

Comment:
Change the use of `deferred_codegen` to be more like what Enzyme expects, e.g. use `::Type{f}` and check that we can now do deferred codegen on closures.